x/swaprouter/router.go

```diff
@@ -130,7 +132,12 @@ func (k Keeper) MultihopEstimateOutGivenExactAmountIn(
 			swapFee = routeSwapFee.Mul((swapFee.Quo(sumOfSwapFees)))
 		}
 
-		tokenOut, err := pool.CalcOutAmtGivenIn(ctx, sdk.Coins{tokenIn}, route.TokenOutDenom, swapFee)
+		poolTrad, ok := pool.(gammtypes.TraditionalAmmInterface)
```

Comment:
Is it possible to make this more general to work with CL pool as well?

We should aim to have everything in `swaprouter` to be generalized for any kind of pool

Reply:
Can we talk about this further? Do you for see just checking if the msg is CL and use the CL keeper method, otherwise use the TraditionalAmmInterface CalcOut?

Reply:
I would instead propagate this logic to the keeper (`swapModule`). Similar to how we route swaps. The keeper methods can deal with that instead